### PR TITLE
Rework diagnostics

### PR DESCRIPTION
**Pull request by olbjo**
With updates included in this PR the first step of the Diagnostics test case passes

### CODE REVIEW

File: src/device/pf_diag.c [hunk 1]
```diff
@@ -100,6 +105,73 @@ static void pf_diag_update_station_problem_indicator (
...
+ */
+static void pf_diag_update_submodule_state (
+   pnet_t * net,
+   pf_ar_t * p_ar,
+   pf_subslot_t * p_subslot)
+{
+   pf_device_t * p_dev = NULL;
+   uint16_t ix;
+   pf_diag_item_t * p_item;
+   /*uint16_t diag_properties;*/
```
Comment (pyhys) [inline src/device/pf_diag.c:124]: Remove line

**Review by pyhys — APPROVED** [review]
Great! Remove a commented line.

File: src/device/pf_diag.c (hunk 1 above)
Comment (olbjo) [reply src/device/pf_diag.c]: Done

**Comment by olbjo** [comment]
A couple of conflicts and build issues had to be fixed. You may want to check out lldp and pdport.

File: src/device/pf_pdport.c [hunk 2]
```diff
@@ -294,6 +294,103 @@ int pf_pdport_read_ind (
...
+ * Run peer check
+ *
+ * Compare llpd peer information with configured pdport peer.
+ * Trigger an alarm if wrong peer is connected.
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param loc_port_num     In:    Local port number.
+ *                                Valid range: 1 .. N, where N is the total
+ *                                number of local ports used by p-net stack.
+ * @param p_lldp_peer_info In:    Peer info
```
Comment (pyhys) [inline src/device/pf_pdport.c:307]: * @param p_lldp_peer_info In:    Currently received peer info

File: src/common/pf_lldp.c [hunk 3]
```diff
@@ -1344,9 +1357,14 @@ int pf_lldp_generate_alias_name (
...
  * Apply updated peer information
+ *
+ * Trigger an alarm if the peer information not is as expected.
+ * Resets the peer watchdog timer.
+ *
  * @param net              InOut: p-net stack instance
  * @param loc_port_num     In:    Local port number.
- *                                Valid range: 1 .. PNET_MAX_PORT
+ *                                Valid range: 1 .. N, where N is the total
+ *                                number of local ports used by p-net stack.
```
Comment (pyhys) [inline src/common/pf_lldp.c:1367]: Keep the old version of the text

**Review by pyhys — COMMENTED** [review]
A few quick-to-fix modifications

File: src/device/pf_pdport.c [hunk 4]
```diff
@@ -294,6 +294,103 @@ int pf_pdport_read_ind (
...
+   int loc_port_num,
+   const pnet_lldp_peer_info_t * p_lldp_peer_info)
+{
+   bool peer_stationname_is_correct = false;
+
+   bool peer_portname_is_correct = false;
+   pnet_diag_source_t diag_source = {0};
+   pf_check_peer_t * p_wanted_peer = &net->port[0].pdport.check.peer;
+
+   /* TODO Handle loc_port_num */
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_pdport.c:320]: done

File: src/device/pf_pdport.c (hunk 2 above)
Comment (olbjo) [reply src/device/pf_pdport.c]: done

File: src/common/pf_lldp.c (hunk 3 above)
Comment (olbjo) [reply src/common/pf_lldp.c]: done

File: src/device/pf_pdport.c [hunk 5]
```diff
@@ -294,6 +294,103 @@ int pf_pdport_read_ind (
...
+            p_lldp_peer_info->chassis_id.len) == 0)
+      {
+         peer_stationname_is_correct = true;
+      }
+
+      // TODO Update also peer_portname_is_correct
+
+      diag_source.api = PNET_API_NO_APPLICATION_PROFILE;
+      diag_source.slot = PNET_SLOT_DAP_IDENT;
+      diag_source.subslot = PNET_SUBSLOT_DAP_INTERFACE_1_PORT_0_IDENT;
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_pdport.c]: done

File: src/device/pf_pdport.c [hunk 6]
```diff
@@ -294,6 +294,103 @@ int pf_pdport_read_ind (
...
+{
+   bool peer_stationname_is_correct = false;
+
+   bool peer_portname_is_correct = false;
+   pnet_diag_source_t diag_source = {0};
+   pf_check_peer_t * p_wanted_peer = &net->port[0].pdport.check.peer;
+
+   /* TODO Handle loc_port_num */
+
+   if (net->port[0].pdport.check.active == true)
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_pdport.c]: done

File: src/common/pf_alarm.c [hunk 7]
```diff
@@ -2086,55 +2086,148 @@ int pf_alarm_close (pnet_t * net, pf_ar_t * p_ar)
...
+         p_alarm_spec->ar_diagnosis = true;
+      }
    }
    else
    {
-      *p_maint_status |= p_diag_item->fmt.std.qual_ch_qualifier;
+      /* Diagnosis in standard format */
+
+      severity_qualifier =
+         (p_diag_item->fmt.std.qual_ch_qualifier & ~0x00000007);
```
Comment (hefloryd) [inline src/common/pf_alarm.c:2144]: Add defines for hardcoded constants.

File: src/common/pf_alarm.c [hunk 8]
```diff
@@ -2086,55 +2086,148 @@ int pf_alarm_close (pnet_t * net, pf_ar_t * p_ar)
...
+         is_appears = true;
+      }
+
+      /* Maintenance qualifier bits */
+      *p_maint_status |= severity_qualifier;
+
+      /* MaintenanceRequired and MaintenanceDemanded bits */
+      if (is_maintenance_required == true)
+      {
+         *p_maint_status |= BIT (0);
```
Comment (hefloryd) [inline src/common/pf_alarm.c:2211]: Add defines for hardcoded constants.

File: src/common/pf_alarm.c (hunk 7 above)
Comment (olbjo) [reply src/common/pf_alarm.c]: done
Also updated pf_diag.c

File: src/common/pf_alarm.c (hunk 8 above)
Comment (olbjo) [reply src/common/pf_alarm.c]: done

**Pull request merged**